Perpetuum.DataDumper/Factory/DataDumper.Resolver.cs

```diff
@@ -0,0 +1,94 @@
+﻿using Open.Nat;
+using Perpetuum.DataDumper.Views;
+using Perpetuum.EntityFramework;
+using Perpetuum.Items.Ammos;
+using Perpetuum.Modules;
+using Perpetuum.Modules.Weapons;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Perpetuum.DataDumper {
+    public partial class DataDumper {
+        public dynamic DumpDataView(DataExportMapping mapping) {
+            Type[] typeArgs = new Type[] { typeof(string) };
+            var reflectedMethod = typeof(DataDumper).GetMethod("DumpDataView", typeArgs);
+            var genericMethod = reflectedMethod.MakeGenericMethod(mapping.ViewType);
+            dynamic results = genericMethod.Invoke(this, new object[] { mapping.Category });
+
+            // No need, just use dynamic
+            //var listType = typeof(List<>).MakeGenericType(mapping.ViewType);
+
+            //var typedResults = Convert.ChangeType(results, listType);
+            return results;
+        }
+
+        public List<T> DumpDataView<T>(string categoryName) where T : EntityDataView {
+            var returnItems = new List<T>();
+
+            // Handle sparks first, refactor later
+            if (typeof(T) == typeof(SparkDataView)) {
+                var sparkData = sparkRepository.GetAll();
+
+                foreach (var spark in sparkData) {
+                    try {
+                        var currentView = NewSparkDataView(spark);
+
+                        returnItems.Add(currentView as T);
+                    } catch (Exception ex) {
+                        Console.WriteLine(ex.Message);
+                    }
+                }
+
+                return returnItems;
+
+            }
+
+            var categoryData = GetDataByItemCategoryName(categoryName);
+
+            foreach (var categoryItem in categoryData) {
+                T currentView;
+
+                string itemName = categoryItem["definitionname"].ToString();
+                int itemId = (int)categoryItem["definition"];
+
+                try {
+                    object currentObject = entityFactory.CreateForData(itemName, EntityIDGenerator.Random);
```

Comment:
Ok, this whole part with the CreateForData and the type checking, I think, could handle typing better where you wouldn't need type-check like is.

Also there is a better idiom I think for checking types *and* casting using *is* or *as* with a variable declaration in line.. its a bit fancy but here are the docs on it: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/type-testing-and-cast